consensus/reactor.go

```diff
@@ -249,6 +249,14 @@ func (conR *Reactor) Receive(chID byte, src p2p.Peer, msgBytes []byte) {
 	case StateChannel:
 		switch msg := msg.(type) {
 		case *NewRoundStepMessage:
+			conR.conS.mtx.Lock()
+			initialHeight := conR.conS.state.InitialHeight
```

Comment:
Isn't this immutable? Maybe we can load it once upfront and then not need to worry about the mutex?

Reply:
Yeah, but I figured that'd be premature optimization unless we could demonstrate that there was actual mutex contention -- complexity (i.e. caching) is usually far more likely to cause problems. Although I suppose storing this during `consensus.NewState()` would pretty much always do the right thing.

Opened #6012, we can have a look after the refactor.

Reply:
Note, this was changed to a read-lock in the consensus reactor p2p refactor